### PR TITLE
Implement more request#26

### PR DESCRIPTION
**Pull request by natthaphongphengsa**
Add function for edit courseMembership

### CODE REVIEW

File: src/EduManagementLab.Api/Controllers/CoursesController.cs [hunk 1]
```diff
@@ -149,6 +149,21 @@ public ActionResult<CourseDto> AddCourseMembership(Guid courseId, Guid userId, D
 
         }
 
+        [HttpPut]
+        [Route("{courseId}/UpdateMembership/{userId}")]
+        public ActionResult<Course> UpdateCourseMembershipEnrolledDate(Guid courseId, Guid userId, UpdateMembershipEnrolledDateModel MembershipEnrolledDate)
```
Comment (martinvehmas) [inline src/EduManagementLab.Api/Controllers/CoursesController.cs:154]: Compile error.
UpdateMembershipEnrolledDateModel is missing.
Should it be a DateTime?

**Review by martinvehmas — CHANGES_REQUESTED** [review]

File: src/EduManagementLab.Api/Controllers/CoursesController.cs [hunk 2]
```diff
@@ -165,6 +180,28 @@ public ActionResult<List<Course.Membership>> GetCourseMemberships(Guid courseId)
             }
         }
 
+        [HttpGet]
+        [Route("{courseId}/Membership/{userId}")]
```
Comment (martinvehmas) [inline src/EduManagementLab.Api/Controllers/CoursesController.cs:184]: Rename the route to "{courseId}/Memberships/{userId}"
Its a collection of memberships.

File: src/EduManagementLab.Api/Controllers/CoursesController.cs [hunk 3]
```diff
@@ -149,6 +149,21 @@ public ActionResult<CourseDto> AddCourseMembership(Guid courseId, Guid userId, D
 
         }
 
+        [HttpPut]
+        [Route("{courseId}/UpdateMembership/{userId}")]
```
Comment (martinvehmas) [inline src/EduManagementLab.Api/Controllers/CoursesController.cs:153]: Rename route to "{courseId}/Memberships/{userId}"
The PUT method tells that the resource should be replaced/updated.

**Review by martinvehmas — CHANGES_REQUESTED** [review]

File: src/EduManagementLab.Api/Controllers/CoursesController.cs (hunk 1 above)
Comment (natthaphongphengsa) [reply src/EduManagementLab.Api/Controllers/CoursesController.cs:154]: yes. It's DatTime.

File: src/EduManagementLab.Api/Controllers/CoursesController.cs (hunk 2 above)
Comment (natthaphongphengsa) [reply src/EduManagementLab.Api/Controllers/CoursesController.cs:184]: I change the Route to {courseId}/Memberships/{userId}

File: src/EduManagementLab.Api/Controllers/CoursesController.cs (hunk 3 above)
Comment (natthaphongphengsa) [reply src/EduManagementLab.Api/Controllers/CoursesController.cs:153]: I rename it to "{courseId}/Memberships/{userId}"

**Review by martinvehmas — APPROVED** [review]

**Pull request merged**